src/stdlib_stats_distribution_exponential.fypp

```diff
@@ -0,0 +1,322 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+module stdlib_stats_distribution_exponential
+    use stdlib_kinds, only : sp, dp, xdp, qp, int32
+    use stdlib_error, only : error_stop
+    use stdlib_random, only : dist_rand
+    use stdlib_stats_distribution_uniform, only : uni=>rvs_uniform
+
+    implicit none
+    private
+
+    integer :: ke(0:255)
+    real(dp) :: we(0:255), fe(0:255)
+    logical  :: zig_exp_initialized = .false.
+
+    public :: rvs_exp
+    public :: pdf_exp
+    public :: cdf_exp
+
+
+
+    interface rvs_exp
+    !! Version experimental
+    !!
+    !! Exponential Distribution Random Variates
+    !! ([Specification](../page/specs/stdlib_stats_distribution_exponential.html#
+    !! rvs_exp-exponential-distribution-random-variates))
+    !!
+        module procedure rvs_exp_0_rsp                 !0 dummy variable
+
+        #:for k1, t1 in RC_KINDS_TYPES
+        module procedure rvs_exp_${t1[0]}$${k1}$       !1 dummy variable
+        #:endfor
+
+        #:for k1, t1 in RC_KINDS_TYPES
+        module procedure rvs_exp_array_${t1[0]}$${k1}$ !2 dummy variables
+        #:endfor
+    end interface rvs_exp
+
+
+
+    interface pdf_exp
+    !! Version experimental
+    !!
+    !! Exponential Distribution Probability Density Function
+    !! ([Specification](../page/specs/stdlib_stats_distribution_exponential.html#
+    !! pdf_exp-exponential-distribution-probability-density-function))
+    !!
+        #:for k1, t1 in RC_KINDS_TYPES
+        module procedure pdf_exp_${t1[0]}$${k1}$
+        #:endfor
+    end interface pdf_exp
+
+
+
+    interface cdf_exp
+    !! Version experimental
+    !!
+    !! Exponential Distribution Cumulative Distribution Function
+    !! ([Specification](../page/specs/stdlib_stats_distribution_exponential.html#
+    !! cdf_exp-exponential-distribution-cumulative-distribution-function))
+    !!
+        #:for k1, t1 in RC_KINDS_TYPES
+        module procedure cdf_exp_${t1[0]}$${k1}$
+        #:endfor
+    end interface cdf_exp
+
+
+
+
+
+contains
+
+    subroutine zigset
```

Comment:
Could it be `pure`?

Reply:
No. The zigset subroutine changes the state of module variable `zig_initialized` and other variables. If we store the table of values accessed by the ziggurat algorithm as module variables, we might eliminate zigset subroutine and make exponential/normal random variables generation functions pure and capable of multithread/multiprocess.